clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java

```diff
@@ -116,9 +116,9 @@
     public void seekToEnd(Collection<TopicPartition> partitions);
 
     /**
-     * @see KafkaConsumer#position(TopicPartition)
+     * @see KafkaConsumer#position(TopicPartition, long)
      */
-    public long position(TopicPartition partition);
+    public long position(TopicPartition partition, long timeout);
```

Comment:
Whether it's a new config or a new API, we'll still need a KIP. If we go that far, we'll want to consider the other methods which probably suffer the same problem (e.g. `seekToBeginning`). A couple options that we have thrown around are the following:

1. Introduce a `max.block.ms` config like the producer has.
2. Overload the methods with timeouts.
3. Use the request timeout.

Not sure the best way to go here to be honest. Maybe for `position()`, we should just raise a `UnknownTopicException` instead of retrying.